control-plane/pkg/autoscaler/keda/resources_test.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package keda
+
+import (
+	"strings"
+	"testing"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/autoscaler/keda/mocks"
+)
+
+func TestGenerateScaledObjectName(t *testing.T) {
+	type args struct {
+		obj v1.Object
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantPrefix string
+		wantSuffix types.UID
+	}{
```

Comment:
Nice use of the table testing pattern! This is how we normally do unit tests in eventing (as I'm sure you noticed)